include/eve/module/core/function/generic/load.hpp

```diff
@@ -0,0 +1,154 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/implementation.hpp>
+#include <type_traits>
+
+namespace eve
+{
+  template<typename T> concept scalar_pointer = requires(T a)
+  {
+    { *a };
+    requires scalar_value<std::remove_cvref_t<decltype(*a)>>;
+  };
+}
+
+namespace eve::detail
+{
+  //================================================================================================
+  // SIMD
+  //================================================================================================
+  template<scalar_pointer Ptr>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), Ptr ptr) noexcept
+                  -> as_wide_t<std::remove_cvref_t<decltype(*std::declval<Ptr>())>>
+  {
+    return as_wide_t<std::remove_cvref_t<decltype(*ptr)>>(ptr);
+  }
+
+  template<scalar_pointer Ptr, typename Cardinal>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), Ptr ptr, Cardinal const &) noexcept
+                  -> as_wide_t<std::remove_cvref_t< decltype(*std::declval<Ptr>())>
+                                                  , typename Cardinal::type
+                                                  >
+  {
+    return as_wide_t<std::remove_cvref_t<decltype(*ptr)>, typename Cardinal::type>(ptr);
+  }
+
+  template<relative_conditional_expr C, scalar_pointer Ptr>
+  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), C const &cond, Ptr ptr) noexcept
+                  -> as_wide_t<std::remove_cvref_t<decltype(*std::declval<Ptr>())>>
+  {
+    using e_t = std::remove_cvref_t<decltype(*ptr)>;
+    using r_t = as_wide_t< e_t >;
+
+    if constexpr( !std::is_pointer_v<Ptr> )
+    {
+      return eve::load[cond](ptr.get());
```

Comment:
We will replace this `load[cond]` with `load_unsafe` when we have it, right?

Reply:
Yes